.github/workflows/test.yml

```
@@ -0,0 +1,227 @@
+name: test
+
+on:
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      rose_ref:
+        description: The Rose branch to test against
+        required: true
+      fcm_ref:
+        description: The FCM branch to test against
+        required: false
+      fcm_repo:
+        description: The FCM branch to test against
+        required: false
+      cylc_ref:
+        description: The Cylc branch to test against
+        required: false
+      cylc_repo:
+        description: The Cylc repo to test against
+        required: false
+
+defaults:
+  run:
+    shell: bash  # macos default shell is zsh
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 45
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ['ubuntu-latest']
+        python-version: ['3.7']
+        # TODO: re-enable macos testing
+        # currently (in the absence of a wrapper script) rose cannot be run
+        # from within cylc jobs in situations where the output of the following
+        # commands differ:
+        #   bash -c 'which python'
+        #   bash -l -c 'which python'
+        # include:
+        #   - os: 'macos-latest'
+        #     python-version: '3.7'
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.inputs.rose_ref || github.sha }}
+          path: rose
+
+      - name: Configure Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Cylc
+        env:
+          cylc_repo: ${{ github.event.inputs.cylc_repo || 'cylc/cylc-flow' }}
+          cylc_branch: ${{ github.event.inputs.cylc_ref || 'master' }}
+        run: |
+          pip install "git+https://github.com/$cylc_repo@$cylc_branch"
```

I think shellchecker would complain about this
```suggestion
          pip install "git+https://github.com/${cylc_repo}@${cylc_branch}"
```

---

Shellcheck is fine with this because `@` is not a valid char for environment variables so `$cylc_repo` cannot be mistaken for `$cylc_repo@`, that said I like braces.